backend/src/main/scala/bloop/Compiler.scala

```diff
@@ -555,9 +578,11 @@ object Compiler {
                   )
                 }
                   .flatMap(clientClassesObserver.nextAnalysis)
-                Task
+
+                deleteBestEffortDir() *> Task
```

Comment:
Isn't this already added below? Or do you need to run it first? In that case we can probably remove it from below. Ideally though, we would remove it together for it to be more efficient.

Also, is there a non symbolic way of defining it? Would that not be just a flatMap?

Reply:
Yes, I need to tun this before `updateClientState` (I missed the repeat below, thank you for the catch!), but I'll try rearranging the tasks so that the other ones can be run concurrently